File: DMOJ/CCC/Maternity.cpp
// CCC '06 S1 - Maternity

#include<bits/stdc++.h>
using namespace std;
int x;
string a, b;
bool genes[6][2];
bool check(string baby){
    int pass = 0;
    for (int i=0; i<10; i+=2){
        if (a[i]==b[i] && baby[i/2]==a[i]) pass++;
        else if (a[i+1]==b[i+1] && baby[i/2]==a[i+1]) pass++;
    }
    return pass==5;
}
int main(){
    cin >> a >> b >> x;
    while (x--){
        string baby; cin >> baby;
        cout << check(baby) << '\n';
    }
}

File: DMOJ/CCC/Maternity.py
# CCC '06 S1 - Maternity

p1, p2 = input(), input()
possible = []
for i in range(0, 10, 2):
    d = min(p1[i:i+2], p2[i:i+2])
    possible.append((not d.islower(), not d.isupper()))

for i in range(int(input())):
    c = input()
    for j in range(5):
        if (c[j].isupper() and not possible[j][0]) or (c[j].islower() and not possible[j][1]):
            print("Not their baby!")
            break
    else:
        print("Possible baby.")